packages/pathfinder/src/pathfinder.ts

```diff
@@ -0,0 +1,400 @@
+        const { 
+            source_amount,
+            source_asset_code, 
+            source_asset_issuer, 
+            source_asset_type,
+            destination_amount,
+            destination_asset_type,
+            destination_asset_issuer,
+            destination_asset_code,
+            path,
+        } = pathFoundResult;
+        const source_asset = (source_asset_type !== 'native') 
+            ? {
+                code: source_asset_code,
+                issuerPublicKey: source_asset_issuer
+            } as stellarAsset
+            : {
+                code: 'XLM',
+                issuerPublicKey: ''
+            } as stellarAsset;
+        const destination_asset = (destination_asset_type !== 'native') 
+            ? {
+                code: destination_asset_code,
+                issuerPublicKey: destination_asset_issuer
+            } as stellarAsset
+            : {
+                code: 'XLM',
+                issuerPublicKey: ''
+            } as stellarAsset;
+        const dest_amount_str = parseFloat(destination_amount);
+        const bucket = this.getBucket(destination_asset, dest_amount_str);
+        const exchangeRate = this.calculateExchangeRate(dest_amount_str, parseFloat(source_amount));
+        const _path = path.map((asset: { asset_code: string, asset_issuer: string, asset_type: string}) => {
+            return { code: asset.asset_code, issuerPublicKey: asset.asset_issuer } as stellarAsset;
+        });
+        return {
+            to: destination_asset,
+            from: source_asset,
+            path: _path,
+            bucket: bucket,
+            exchangeRate: exchangeRate
+        } as IPathInfo;
+    }    
+     
+    /**
+     * Helper For below
+     * Creates a StellarSdk.Asset to be used in server calls
+     */
+    private constructStellarAsset = (asset: stellarAsset): StellarSdk.Asset => {
+        const { code, issuerPublicKey } = asset;
+        const returnAsset = (code !== 'X:M') 
+            ? new StellarSdk.Asset(code, issuerPublicKey)
+            : StellarSdk.Asset.native();
+        return returnAsset;
+    }
+    
+    /**
+     * Returns a list of the best paths from each of the supported assets to the given
+     * destination asset and amount 
+     * 
+     * Note: uses REFERENCE_ACCOUNT_PUBLIC_KEY as the destination and source accounts
+     */
+    private async getBestPaths(
+        destinationAsset: stellarAsset,
+        destinationAmount: number,
+    ): Promise<IPathInfo[]> {
+
+        const _destinationAsset = this.constructStellarAsset(destinationAsset);
+        return this.server.paths(Pathfinder.REFERENCE_ACCOUNT_PUBLIC_KEY, 
+                            Pathfinder.REFERENCE_ACCOUNT_PUBLIC_KEY, 
+                            _destinationAsset, String(destinationAmount))
+                    .call()
+                    .then((paths: any) => JSON.parse(JSON.stringify(paths)).records)
+                    .then((paths: StellarSdk.PaymentPathRecord[]) => {
```

Comment:
Could we add just some brief comments here describing the logical sections of this code block?

Reply:
added